src/main/java/io/github/syst3ms/skriptparser/expressions/ExprStringCharAt.java

```diff
@@ -0,0 +1,53 @@
+package io.github.syst3ms.skriptparser.expressions;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import io.github.syst3ms.skriptparser.Parser;
+import io.github.syst3ms.skriptparser.lang.Expression;
+import io.github.syst3ms.skriptparser.lang.TriggerContext;
+import io.github.syst3ms.skriptparser.parsing.ParseContext;
+import io.github.syst3ms.skriptparser.util.DoubleOptional;
+
+/**
+ * The character at a position of a string. The first character is at the position 0.
+ *
+ * @name Character at
+ * @pattern char[acter] at [(index|pos[ition])] %number% (of|in) %string%
+ * @since ALPHA
+ * @author Olyno
+ */
+public class ExprStringCharAt implements Expression<String> {
+
+	static {
+		Parser.getMainRegistration().addExpression(
+			ExprStringCharAt.class,
+			String.class,
+			true,
+			"char[acter] at [(index|pos[ition])] %integer% (of|in) %string%"
+		);
+	}
+
+	private Expression<Integer> position;
+	private Expression<String> value;
+
+	@SuppressWarnings("unchecked")
+	@Override
+	public boolean init(final Expression<?> @NotNull [] expressions, final int matchedPattern, @NotNull final ParseContext parseContext) {
+		position = (Expression<Integer>) expressions[0];
+		value = (Expression<String>) expressions[1];
+		return true;
+	}
+
+	@Override
+	public String[] getValues(final TriggerContext ctx) {
+		DoubleOptional<? extends String, ? extends Integer> opt = DoubleOptional.ofOptional(value.getSingle(ctx), position.getSingle(ctx));
+		return opt.mapToOptional((val, pos) -> new String[]{ String.valueOf(val.charAt(pos)) } )
```

Comment:
```suggestion
		return opt.mapToOptional((val, pos) -> new String[]{ String.valueOf(val.charAt(pos - 1)) } )
```
1. Indices in Skript start at 1 (everywhere, so we should keep consistency in my opinion, even when it seems odd)
2. `String#charAt()` throws an IndexOutOfBoundException you should handle. You can do this by filtering the optional to only contain the value if the index is within the bounds of the String (obviously also with the 'minus 1').